server/src/main/java/org/kaazing/gateway/server/util/version/DuplicateJarFinder.java

```diff
@@ -0,0 +1,113 @@
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.kaazing.gateway.server.util.version;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.jar.Attributes;
+
+import org.slf4j.Logger;
+
+/**
+ * Class used for detecting if duplicate jar files are loaded by the gateway.
+ */
+public class DuplicateJarFinder {
+
+    private static final String JAR_FILE_WITH_VERSION_LOGGING_MESSAGE = "The jar file {} with the version {} was loaded.";
+    private static final String CONFLICTING_JARS_LOGGING_MESSAGE =
+            "Failed to start because of conflicting versions of artifact on the classpath: {}.jar.";
+    private static final String MANIFEST_VERSION = "Implementation-Version";
+    private static final String MANIFEST_JAR_NAME = "Jar-Name";
+    private static final String MANIFEST_PRODUCT = "Kaazing-Product";
+
+    private ClassPathParser classPathParser;
+    private Logger gatewayLogger;
+    private Set<String> duplicateJars = new HashSet<String>();
+    private Set<String> loadedJars = new HashSet<String>();
+
+    public DuplicateJarFinder(Logger gatewayLogger) {
+        this.gatewayLogger = gatewayLogger;
+        this.classPathParser = new ClassPathParser();
+    }
+
+    /**
+     * Parses the class path system attribute and the manifest files
+     * and if there are duplicate jar a DuplicateJarsException is thrown.
+     * @throws IOException
+     * @throws DuplicateJarsException
+     */
+    public void findDuplicateJars() throws IOException, DuplicateJarsException {
+        String[] classPathEntries = classPathParser.getClassPathEntries();
+        for (String classPathEntry : classPathEntries) {
+            parseManifestFileFromClassPathEntry(classPathEntry);
+        }
+        logErrorForDuplicateJars();
+        throwExceptionIfDuplicateJarsAreFound();
+    }
+
+    private void parseManifestFileFromClassPathEntry(String classPathEntry) throws IOException {
+        Attributes manifestAttributes = classPathParser.getManifestAttributesFromClassPathEntry(classPathEntry);
+        String product = manifestAttributes.getValue(MANIFEST_PRODUCT);
```

Comment:
I don't immediately see the need to for having to check whether it is a Kaazing Product or not, this would also make your like simpler with the jar plugin.  Is there some jars that you hit that we have multiple of in testing?  

Another option would be to put the group Id and artifact Id in the jar name, and then check if the group Id or artifact Id has "kaazing" in it.  In fact after thinking about this, I think this would be better then additing a Kaazing-Product entry to the manifest.  The reason being that jars in the lib directory have com.kaazing or org.kaazing in front of them so it will be easier for customers to identify, and there may be a org.kaazing and com.kaazing jar that has the same artifactId